datafusion/common/src/config.rs

```diff
@@ -973,16 +980,24 @@ impl<F: ConfigField + Default> ConfigField for Option<F> {
 
 #[macro_export]
 macro_rules! config_field {
-    ($t:ty) => {
+    ($t:ty $(, $transform:expr)?) => {
         impl ConfigField for $t {
             fn visit<V: Visit>(&self, v: &mut V, key: &str, description: &'static str) {
                 v.some(key, self, description)
             }
 
             fn set(&mut self, _: &str, value: &str) -> Result<()> {
+                $(
+                    let value = $transform(&value);
```

Comment:
Can the transform and parse be one step?

if 'transform' is defined, this is what gets converts string input to desired value
else the 'value.parse()` is used.

this would allow more naturally express input-to-value transformation without having to define "normalized string format".

wdyt?